clang/lib/Basic/Targets/SystemZ.cpp

```diff
@@ -11,6 +11,7 @@
 //===----------------------------------------------------------------------===//
 
 #include "SystemZ.h"
+#include "clang/AST/Decl.h"
```

Comment:
I think we should not depend on `clangAST` here.

Reply:
What is the motivation for this? I mean, is there any general consensus that this really should be avoided for the purpose of building the tests (note that clang itself builds fine)?  Why did you remove clangAST from the cmake file to begin with - was there some discussion preceding this?

The reason I am passing the VarDecl to the target is that in order to have the (systemz) target return a minimum alignment value it needs to know things about the Variable. Since this is very target specific, it would be a bit weird to have in common code something like isVarExternalOrWeak(), or would it not? AFIK this is only relevant for SystemZ so it would be best to only do this check in the target. Besides, if some other target needs to check something else, it could do so if the *VD is there.

@efriedma-quic do you have any opinion?

Reply:
If I recall correctly, in certain build configurations, like if you enable shared libraries, circular dependencies in libraries can cause link errors.  So we structure code to avoid them.

I'd suggest just splitting getMinGlobalAlign into two functions: getMinGlobalAlign() for all globals, and getMinGlobalAlignDefinition() that only applies to variable definitions.  Then you can put the VarDecl checks into the ASTContext code.

Reply:
Ok, that makes more sense to me know considering different ways of building clang, thanks.

Patch updated: I tried to follow your suggestion by splitting up the work and basically inspecting the VarDecl in ASTContext and then pass only a boolean to TargetInfo.  Is this something like what you had in mind?

Reply:
Thanks. Now you can remove this. This is the only point from my side.

> (note that clang itself builds fine)
You were introducing layering violation. "Clang could be built apparently."

> Why did you remove clangAST from the cmake file to begin with - was there some discussion preceding this?
In principle, we can prune redundant deps. In contrast, the discussion would be required for introducing new deps.

Reply:
Ah, one more #include to remove that I missed - thanks :)

Reply:
Thanks!